contracts/multisigs/GuardCM.sol

```diff
@@ -0,0 +1,292 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import {Enum} from "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+interface IGovernor {
+    function state(uint256 proposalId) external returns (ProposalState);
+}
+
+// Governor proposal state
+enum ProposalState {
+    Pending,
+    Active,
+    Canceled,
+    Defeated,
+    Succeeded,
+    Queued,
+    Expired,
+    Executed
+}
+
+/// @dev Only `owner` has a privilege, but the `sender` was provided.
+/// @param sender Sender address.
+/// @param owner Required sender address as an owner.
+error OwnerOnly(address sender, address owner);
+
+/// @dev Only `manager` has a privilege, but the `sender` was provided.
+/// @param sender Sender address.
+/// @param manager Required sender address as an owner.
+error ManagerOnly(address sender, address manager);
+
+/// @dev Provided zero address.
+error ZeroAddress();
+
+/// @dev Provided zero value.
+error ZeroValue();
+
+/// @dev Wrong length of two arrays.
+/// @param numValues1 Number of values in a first array.
+/// @param numValues2 Numberf of values in a second array.
+error WrongArrayLength(uint256 numValues1, uint256 numValues2);
+
+/// @dev Provided incorrect data length.
+/// @param expected Expected minimum data length.
+/// @param provided Provided data length.
+error IncorrectDataLength(uint256 expected, uint256 provided);
+
+/// @dev No delegatecall is allowed.
+error NoDelegateCall();
+
+/// @dev No self multisig call is allowed.
+error NoSelfCall();
+
+/// @dev The combination of target and selector is not authorized.
+/// @param target Target address.
+/// @param selector Function selector.
+error NotAuthorized(address target, bytes4 selector);
+
+/// @dev The proposal is not defeated.
+/// @param proposalId Proposal Id.
+/// @param state Current proposal state.
+error NotDefeated(uint256 proposalId, ProposalState state);
+
+/// @title GuardCM - Smart contract for Gnosis Safe community multisig (CM) guard
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+contract GuardCM {
+    event GovernorUpdated(address indexed governor);
+    event SetTargetSelectors(address[] indexed targets, bytes4[] indexed selectors, bool[] statuses);
+    event GovernorCheckProposalIdChanged(uint256 indexed proposalId);
+    event GuardPaused(address indexed account);
+    event GuardUnpaused();
+
+    // Owner address
+    address public immutable owner;
```

Comment:
what if we need to update the Timelock? Can we make this mutable via owner?

Reply:
If yes, change owner method has to be added

Reply:
We do not believe there is a scenario where we need to update the Timelock. If that's the case, we need to re-deploy and re-set the guard. This would involve changing the module as well.

Reply:
In that case we need also to have a governance vote that pause guard and remove the timelock module form the CM then